scrapers_next/nd/people.py

```diff
@@ -1,118 +1,160 @@
-            p.district_office.address = address
-        except SelectorError:
-            pass
+        contact_patterns = ["Cell", "Fax", "Home", "Work", "Email"]
+        collected_contacts = {}
 
-        try:
-            phones = XPath(
-                "//*[@id='block-system-main']//div[contains(text(), 'phone')]"
-            ).match(self.root)
-            for phone in phones:
-                phone_type = phone.text_content().strip()
-                phone_number = phone.getnext().text_content().strip()
-                if phone_type == "Cellphone:":
-                    p.extras["Cell phone"] = phone_number
-                elif phone_type == "Home Telephone:":
-                    p.extras["Home phone"] = phone_number
-                elif phone_type == "Office Telephone:":
-                    p.district_office.voice = phone_number
-        except SelectorError:
-            pass
+        for pattern in contact_patterns:
+            pattern_match = re.search(rf"{pattern}\s+\S+", contact_node)
+            if pattern_match:
+                field = contact_node[pattern_match.start() : pattern_match.end()]
+                contact = field.split()[1]
+                collected_contacts[pattern] = contact
+            else:
+                collected_contacts[pattern] = ""
 
-        try:
-            email = (
-                XPath("//*[@id='block-system-main']//div[contains(text(), 'Email')]")
-                .match_one(self.root)
-                .getnext()
-                .text_content()
-                .strip()
-            )
-        except SelectorError:
-            email = ""
-        p.email = email
+        p = ScrapePerson(
+            name=self.input.name,
+            given_name=given_name,
+            family_name=family_name,
+            state="nd",
+            image=self.input.image,
+            chamber=self.input.chamber,
+            party=self.input.party,
+            district=self.input.district,
+            email=collected_contacts["Email"],
+        )
 
-        try:
-            fax = (
-                XPath("//*[@id='block-system-main']//div[contains(text(), 'Fax')]")
-                .match_one(self.root)
-                .getnext()
-                .text_content()
-                .strip()
-            )
-            p.district_office.fax = fax
-        except SelectorError:
-            pass
+        p.district_office.address = address
+        p.district_office.voice = collected_contacts["Work"]
+        p.district_office.fax = collected_contacts["Fax"]
+
+        for phone_type in ("Cell", "Home"):
+            if collected_contacts.get(phone_type):
+                p.extras[phone_type.lower()] = collected_contacts[phone_type]
+
+        p.add_source(str(self.source), "member details page")
+
+        members_list_url = str(self.input.url)
+        p.add_source(members_list_url, "members list page")
+
+        url_match = re.search(r".+members", members_list_url)
```

Comment:
Just one list page and the member detail page would be enough, but if you want to have it be the cleaner looking first page url, go for it!

Reply:
I took your guidance and [deleted those lines](https://github.com/openstates/openstates-scrapers/pull/4162/commits/ed81cdcef99f666fd2d8464c036fe5263b687dee#diff-b4eb1b6b6189a33b171b390670cb2f7c25db0b25acd7e596c8c5afb685d778a6L78-L81) in the new commit.